internal/vds/vds_test.go

```diff
@@ -425,6 +425,61 @@ func TestFence(t *testing.T) {
 	}
 }
 
+func TestFenceBorders(t *testing.T) {
+	testcases := []struct {
```

Comment:
The benefit of having a name to a test is that you can explain the contexts of the numbers you are using a bit better. 

E.g. "outside lower bound", "outside upper bound"

And maybe you already do, but it's a bit hard to tell: it makes sense to test the "just out-of-bound" cases. E.g. for index in 'i' direction that would be `-0.49` (lower) and `2.5` (upper)